src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -11494,5 +11495,671 @@ .maxstack  0
+ @"<>f__AnonymousDelegate0
+<>f__AnonymousDelegate0");
+
+            var m = ModuleMetadata.CreateFromImage(verifier.EmittedAssemblyData);
+            var reader = m.MetadataReader;
+            CheckNames(reader, reader.GetTypeDefNames(), "<Module>", "<>f__AnonymousDelegate0", "Program", "<>c");
+        }
+
+        [Fact]
+        public void LambdaWithIdenticalSignatureOptionalMismatch()
+        {
+            var source = """
+using System;
+
+class Program
+{
+    public static void Report(object obj) => Console.WriteLine(obj.GetType()); 
+    public static void Main()
+    {
+        var lam1 = (int x = 10) => x + x;
+        var lam2 = (int x) => x + 1;
+        Report(lam1);
+        Report(lam2);
+    }
+}
+""";
+            var verifier = CompileAndVerify(source, expectedOutput:
+@"<>f__AnonymousDelegate0
+System.Func`2[System.Int32,System.Int32]
+");
+
+            var m = ModuleMetadata.CreateFromImage(verifier.EmittedAssemblyData);
+            var reader = m.MetadataReader;
+            CheckNames(reader, reader.GetTypeDefNames(), "<Module>", "<>f__AnonymousDelegate0", "Program", "<>c");
+        }
+
+        [Fact]
+        public void LambdaIdenticalArityIdenticalDefaultParamDifferentRequiredParams()
+        {
+            var source = """
+using System;
+
+class Program
+{
+    public static void Report(object obj) => Console.WriteLine(obj.GetType()); 
+    public static void Main()
+    {
+        var lam1 = (double d, int x = 10) => { };
+        var lam2 = (string s, int x = 10) => { };
+        Report(lam1);
+        Report(lam2);
+    }
+}
+""";
+
+            // PROTOTYPE: these two delegate types could be unified into a single anonymous delegate class that has a type parameter
+            // for the first argument, so the invoke would look like `<>f__AnonymousDelegate0<T>::Invoke(T arg1, int arg2 = 10)`.
+            // Do we want to change the behavior here to avoid generating too many anonymous delegates?
+            CompileAndVerify(source, expectedOutput:
+@"<>f__AnonymousDelegate0
+<>f__AnonymousDelegate1");
+        }
+
+        [Fact]
+        public void LambdaConversionDefaultParameterValueMismatch()
+        {
+            var source = """
+using System;
+
+class Program
+{
+    public static void Report(object obj) => Console.WriteLine(obj.GetType()); 
+    public static void Main()
+    {
+        var lam1 = (int x = 10) => x + x;
+        var lam2 = (int x = 20) => x + 1;
+        lam1 = lam2;
+        lam1();
+    }
+}
+""";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (10,16): error CS0029: Cannot implicitly convert type '<anonymous delegate>' to '<anonymous delegate>'
+                //         lam1 = lam2;
+                Diagnostic(ErrorCode.ERR_NoImplicitConv, "lam2").WithArguments("<anonymous delegate>", "<anonymous delegate>").WithLocation(10, 16));
+        }
+
+        [Fact]
+        public void LambdaNoConversionDefaultParameterValueMatch()
```

Comment:
It would likely better communicate what you were going for by calling this `LambdaDefaultParameterNameMismatch`.